internal/juju/kubernetesClouds.go

```diff
@@ -0,0 +1,279 @@
+}
+
+// ReadKubernetesCloud reads a Kubernetes cloud with juju cloud facade.
+func (c *kubernetesCloudsClient) ReadKubernetesCloud(input ReadKubernetesCloudInput) (*ReadKubernetesCloudOutput, error) {
+	conn, err := c.GetConnection(nil)
+	if err != nil {
+		return nil, err
+	}
+	defer func() { _ = conn.Close() }()
+
+	kubernetesAPIClient := c.getKubernetesCloudAPIClient(conn)
+
+	cld, err := kubernetesAPIClient.Cloud(names.NewCloudTag(input.Name))
+	if err != nil {
+		return nil, errors.Annotate(err, "getting clouds")
+	}
+
+	userName := getCurrentJujuUser(conn)
+
+	cloudCredentialTags, err := kubernetesAPIClient.UserCredentials(names.NewUserTag(userName), names.NewCloudTag(input.Name))
+	if err != nil {
+		return nil, errors.Annotate(err, "getting user credentials")
+	}
+	if len(cloudCredentialTags) == 0 {
+		return nil, errors.NotFoundf("cloud credentials for user %q", userName)
+	}
+
+	credentialName := cloudCredentialTags[0].Name()
+
+	parentCloudName, parentCloudRegion := getParentCloudNameAndRegion(cld.HostCloudRegion)
+	return &ReadKubernetesCloudOutput{
+		Name:              input.Name,
+		CredentialName:    credentialName,
+		ParentCloudName:   parentCloudName,
+		ParentCloudRegion: parentCloudRegion,
+	}, nil
+}
+
+// getParentCloudNameAndRegion returns the parent cloud name
+// and region from the host cloud region. HostCloudRegion represents the k8s
+// host cloud. The format is <cloudName>/<region>.
+func getParentCloudNameAndRegion(hostCloudRegion string) (string, string) {
+	parts := strings.Split(hostCloudRegion, "/")
+	if len(parts) != 2 {
+		return "", ""
+	}
+	return parts[0], parts[1]
+}
+
+// UpdateKubernetesCloud updates a Kubernetes cloud with juju cloud facade.
+func (c *kubernetesCloudsClient) UpdateKubernetesCloud(input UpdateKubernetesCloudInput) error {
+	conn, err := c.GetConnection(nil)
+	if err != nil {
+		return err
+	}
+	defer func() { _ = conn.Close() }()
+
+	kubernetesAPIClient := c.getKubernetesCloudAPIClient(conn)
+
+	conf, err := clientcmd.NewClientConfigFromBytes([]byte(input.KubernetesConfig))
+	if err != nil {
+		return errors.Annotate(err, "parsing kubernetes configuration data")
+	}
+
+	apiConf, err := conf.RawConfig()
+	if err != nil {
+		return errors.Annotate(err, "fetching kubernetes configuration")
+	}
+
+	var k8sContextName string
+	if input.KubernetesContextName == "" {
+		k8sContextName = apiConf.CurrentContext
+	} else {
+		k8sContextName = input.KubernetesContextName
+	}
+
+	var hostCloudRegion string
+	if input.ParentCloudName != "" || input.ParentCloudRegion != "" {
+		hostCloudRegion = input.ParentCloudName + "/" + input.ParentCloudRegion
+	} else {
+		hostCloudRegion = k8s.K8sCloudOther
+	}
+
+	newCloud, err := k8scloud.CloudFromKubeConfigContext(
+		k8sContextName,
+		&apiConf,
+		k8scloud.CloudParamaters{
+			Name:            input.Name,
+			HostCloudRegion: hostCloudRegion,
+		},
+	)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	err = kubernetesAPIClient.UpdateCloud(newCloud)
```

Comment:
I'm wondering if the created credential will need an update too?

Reply:
Hmm... and what is your suggestion? to take credentials from the state and compare?